packages/core/src/account/types.ts

```diff
@@ -55,7 +55,7 @@ export interface ISmartContractAccount<
    *
    * @returns a dummy signature that doesn't cause the account to revert during estimation
    */
-  getDummySignature(): Hex;
+  getDummySignature(): Promise<Hex>;
```

Comment:
oh right I forgot we were doing this... tbf I think this mostly gets used internally so we could just push it as a non-breaking change 😬  

that feels wrong to say, but doing a whole update to v4 just for this also feels wrong. If we want to update to v4 then @adam-alchemy we might as well break the apis for `createModularAccount` rather than deprecating them

We could also move the account definition and plugin defs into aa-alchemy instead which means we can remove the need for 2 different plugin gen implementations